algorithms/kernel/kernel_function/oneapi/kernel_function_linear_dense_default_kernel_oneapi.h

```diff
@@ -0,0 +1,74 @@
+/* file: kernel_function_linear_dense_default_kernel_oneapi.h */
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Declaration of template structs that calculate SVM Linear Kernel functions.
+//--
+*/
+
+#ifndef __KERNEL_FUNCTION_DENSE_LINEAR_KERNEL_ONEAPI_H__
+#define __KERNEL_FUNCTION_DENSE_LINEAR_KERNEL_ONEAPI_H__
+
+#include "algorithms/kernel/kernel_function/oneapi/kernel_function_linear_base_oneapi.h"
+
+namespace daal
+{
+namespace algorithms
+{
+namespace kernel_function
+{
+namespace linear
+{
+namespace internal
+{
+using namespace daal::data_management;
+using namespace daal::services;
+
+template <typename algorithmFPType>
+class KernelImplLinearOneAPI<defaultDense, algorithmFPType> : public Kernel
+{
+public:
+    services::Status compute(ComputationMode computationMode, NumericTable & a1, NumericTable & a2, NumericTable & r,
+                             const daal::algorithms::Parameter * par)
+    {
+        const ParameterBase * svmPar = static_cast<const ParameterBase *>(par);
+
+        switch (computationMode)
+        {
+        case vectorVector: return computeInternalVectorVector(a1, a2, r, svmPar); break;
+        case matrixVector: return computeInternalMatrixVector(a1, a2, r, svmPar); break;
+        case matrixMatrix: return computeInternalMatrixMatrix(a1, a2, r, svmPar); break;
+        }
+
+        DAAL_ASSERT(false); //should never come here
+        return services::Status();
```

Comment:
Probably, use `default` switch branch for that? And set appropriate error code inside status.

Reply:
@michael-smirnov, having an assert after the switch is common practice in such cases. All the errors should be caught in compile time, runtime error handling is not needed at all. The right option is to have `static_assert`, since `DAAL_ASSERT` is compiled only in debug mode.

Reply:
Ok. I return status error, if `computationMode` set incorrect.